jsre/natto.go

```diff
@@ -0,0 +1,170 @@
+/*
+
+This file is a modified version of https://github.com/robertkrimen/natto/blob/master/natto.go
+
+Package natto is an example/offshoot of otto that implements an event loop (supporting setTimeout/setInterval).
+
+http://godoc.org/github.com/robertkrimen/natto
+
+otto: http://github.com/robertkrimen/otto (http://godoc.org/github.com/robertkrimen/otto)
+
+// The VM has the following functions available:
+//
+//      <timer> = setTimeout(<function>, <delay>, [<arguments...>])
+//      <timer> = setInterval(<function>, <delay>, [<arguments...>])
+//      clearTimeout(<timer>)
+//      clearInterval(<timer>)
+//
+
+*/
+package jsre
+
+import (
+	"github.com/robertkrimen/otto"
+	"time"
+)
+
+type nattoTimer struct {
+	timer    *time.Timer
+	duration time.Duration
+	interval bool
+	call     otto.FunctionCall
+}
+
+type evalReq struct {
+	code   interface{}
+	result otto.Value
+	err    error
+	done   chan bool
+}
+
+type natto struct {
+	vm      *otto.Otto
+	eval    chan *evalReq
+	stop    chan bool
+	running bool
+}
+
+func newNatto(vm *otto.Otto) (self *natto) {
```

Comment:
It is kind-of pointless to have this as a separate type. Please move the logic into `JSRE`.